xgcm/test/test_metrics.py

```diff
@@ -311,28 +307,94 @@ def test_set_metric():
         assert k in grid_manual._metrics.keys()
 
         for metric_expected, metric in zip(v, grid_manual._metrics[k]):
-            xr.testing.assert_equal(metric_expected.reset_coords(drop=True), metric)
+            xr.testing.assert_allclose(metric_expected.reset_coords(drop=True), metric)
 
         for metric_expected, metric in zip(v, grid._metrics[k]):
-            xr.testing.assert_equal(metric_expected.reset_coords(drop=True), metric)
+            xr.testing.assert_allclose(metric_expected.reset_coords(drop=True), metric)
 
 
-@pytest.mark.skip(reason="For the next PR")
 @pytest.mark.parametrize(
-    "metric_axes,metric_name",
+    "metric_axes, metric_varname, overwrite_",
     [
-        ("X", "dx_t"),
-        ("Y", "dy_ne"),
+        ("X", ["dx_n", "dx_t", "dx_ne", "dx_t"], True),
+        ("Y", ["dy_n", "dy_t", "dy_ne", "dy_t"], True),
+        (("X", "Y"), ["area_n", "area_t", "area_ne", "area_t"], True),
+        # ("X", [("dx_n", "dx_t"),("dx_t", "dx_ne")], True),
+        pytest.param(
+            "X",
+            ["dx_n", "dx_t", "dx_ne", "dx_t"],
+            False,
+            marks=pytest.mark.xfail,
+            id="failed to overwrite same metric_varname",
+        ),
     ],
 )
-def test_interp_metrics(metric_axes, metric_name):
-    # need to test with boundary and fill_value conditions
+def test_set_metric_overwrite(metric_axes, metric_varname, overwrite_):
     ds, coords, metrics = datasets_grid_metric("C")
     grid = Grid(ds, coords=coords)
-    grid.set_metrics(metric_axes, metric_name)
-    interp_metric = grid._interp_metric(ds.u, metric_axes)
 
-    expected_metric = grid.interp(ds[metric_name], metric_axes)
+    for mv in metric_varname:
+        grid.set_metrics(metric_axes, mv, overwrite=overwrite_)
```

Comment:
Is this actually overwriting anything? I dont think you are setting the metric before?

To really test that a new dataarray is set I would do the following in the test:
1) Set all metrics that are in the dataset with `grid=Grid(ds, ..., metrics=metrics)`
2) Modify a metric and assign it under a new name. For example:
```python
metric_varname_modified = f'{metric_varname}_overwrite'
ds = ds.assign_coords(metric_varname_modified, ds[metric_varname]*10)
```
3) Then assign this modified metric to the dataset with `grid.set_metric(metric_axes, metric_varname_modified)`. You will have to write this in a loop, this is just an example. 
4) Now you can check explicitly that the new metric is actually 10 times as large. Without that you cant be really sure that the metric has been replaced!

Reply:
Hi, I was trying to incorporate this into the tests, and I wanted to clarify. Is the purpose of the test to:
1. Add a new metric that is 10x the value of the old one (e.g., you have both `dx_t` and `dx_t_overwrite` as metrics)
2. Replace an existing metric with something named similarly but 10x in value (e.g., you end up with one `dx_t` metric that's 10x in value as the original one from metrics=metrics)

Reply:
Yes that way you can be sure that what you are checking is not just the metric set in the first place.